src/Security/Authentication/Negotiate/src/PostConfigureNegotiateOptions.cs

```diff
@@ -59,6 +61,34 @@ public void PostConfigure(string name, NegotiateOptions options)
                         + " Enable Windows Authentication for the server and the Negotiate Authentication handler will defer to it.");
                 }
             }
+
+            var ldapOptions = options.LdapOptions;
+
+            if (ldapOptions.EnableLdapRoleClaimResolution)
+            {
+                if (ldapOptions.LdapConnection == null)
+                {
+                    var di = new LdapDirectoryIdentifier(server: ldapOptions.Domain, fullyQualifiedDnsHostName: true, connectionless: false);
```

Comment:
Confirm that NegotiateOptions.Validate runs before this. I think it might run afterwards, leaving you with invalid values here.

Reply:
Hmm I remember verifying this earlier but I'll double check that this is still the case.

Reply:
Ah you're right, post configure runs before validate.

Reply:
I'll just call validate on LdapSettings during post-configure.